### PR TITLE
Autocomplete added to work form for keywords

### LINKED ISSUE #264: Add autocomplete for keywords
These should come from OCLC FAST authority hosted by Cornell's QA.

See conversation in #82

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

It provides suggestions when users type in the keyword text box in the work form.

image

Closes #264 

## How was this change tested?

locally, so far and unit tests for the autocomplete_controller.

### CODE REVIEW

File: app/views/autocomplete/show.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<% @suggestions.each_with_index.map do |suggestion, i| %>
```
Comment (jcoyne) [inline app/views/autocomplete/show.erb:1]: Is this file used by anything?

File: spec/requests/autocomplete_spec.rb [hunk 2]
```diff
@@ -11,48 +11,73 @@
...
+        'Broncobusters',
+        'Information brokers',
+        'New York (State)--New York--Brooklyn'
+      ]
+    end
     let(:lookup_resp_body) do
       <<~JSON
-        [
-          {
-            "uri": "http://id.worldcat.org/fast/170115",
```
Comment (jcoyne) [inline spec/requests/autocomplete_spec.rb:21]: I believe we want to record the URI too. Does this still permit that?

File: app/views/autocomplete/show.erb (hunk 1 above)
Comment (ndushay) [reply app/views/autocomplete/show.erb:1]: yes - it's how the autocomplete#show command takes the suggestions and serves them up the way the stimulus-autocomplete plugin needs.

Comment (jcoyne) [reply app/views/autocomplete/show.erb:1]: I see that conditional that would render this. Should it have an `html.erb` suffix?

Comment (ndushay) [reply app/views/autocomplete/show.erb:1]: It didn't work with the html.erb suffix.

It's not html, it's an html fragement, so I typed it as "text".   If it were typed as html, but then I think we'd get the full html headers and such.

Comment (jcoyne) [reply app/views/autocomplete/show.erb:1]: Did you try passing `layout: false`?

File: spec/requests/autocomplete_spec.rb (hunk 2 above)
Comment (ndushay) [reply spec/requests/autocomplete_spec.rb:21]: The best we can do is get idroot back from the request https://www.oclc.org/developer/develop/web-services/fast-api/assign-fast.en.html

We can get "idroot" back from autosuggest, and then manipulate it, but it's not always available (maybe only for subjects)  https://fast.oclc.org/searchfast/fastsuggest?callback=&query=mcrae&queryIndex=suggestall&queryReturn=suggestall,idroot&rows=10

image

From their own examples, there seems to be a "suggest" param, but it's not documented:

`suggest=autosuggest` from their first example on their doc page doesn't work, and `suggest=autoSubject` does work but has different results from leaving it out.

File: app/views/autocomplete/show.erb (hunk 1 above)
Comment (ndushay) [reply app/views/autocomplete/show.erb:1]: I don't understand - would `layout : false` be in the controller?   

Are you suggesting I put a method in the controller to do the reformatting?  Or that I add `layout : false` and then use `html.erb` as the suffix?   Maybe a realtime conversation would be better.

File: spec/requests/autocomplete_spec.rb (hunk 2 above)
Comment (jcoyne) [reply spec/requests/autocomplete_spec.rb:21]: @ndushay would the Linked Data API have what we need? https://www.oclc.org/developer/develop/web-services/fast-api/linked-data.en.html

Comment (jcoyne) [reply spec/requests/autocomplete_spec.rb:21]: @arcadia Do you have ideas about what we ought to do when there is no identifier for the topic in FAST?

Comment (ndushay) [reply spec/requests/autocomplete_spec.rb:21]: @arcadiafalcone ^^

Comment (arcadiafalcone) [reply spec/requests/autocomplete_spec.rb:21]: The ones without identifiers are alternate labels and have identifiers under the main label -- for example, "McRae, Col." is an alternate label for "McCrae, John, 1872-1918," which does have an identifier: http://id.worldcat.org/fast/170115. So there may need to be a second lookup to get the main label and identifier. Though that may be weird for the user, as the term will be different from what they select. I suspect that ideally we would only present the main terms to the user to reduce confusion, but I don't know if that's possible.

**Comment by ndushay** [comment]
@jcoyne I already had to change the description generator spec, because I had to change the description generator to get this working.  Do we still need a separate ticket for adding tests for this?

**Comment by amyehodge** [comment]
This looks good to me.

File: app/views/autocomplete/show.erb (hunk 1 above)
Comment (jcoyne) [reply app/views/autocomplete/show.erb:1]: Yes. You should have a `render layout: false` in the controller and then rename this with an `html.erb` suffix.

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Review by jcoyne — APPROVED** [review]

**Pull request merged**